Feature/dll build script

gwyddion39 opened the pull request.

gwyddion39 commented:
@egorpushkin PTAL

This creates a zip with the Android/iOS/Shared Dlls and a version.properties file. The version file contains the Xamarin SDK number and EnRoute Client SDK number.

egorpushkin commented:
Lets create some structure inside that package. Having few DLLs in the root is not clean enough. Replicate structure of our existing SDK package. introduce /libs /source /etc whatever folders you need. 

Rename script to build_xamarin.sh. 

Does it make sense for us to split iOS, Android into separate packages? What are the pros/cons?

egorpushkin commented:
PAC

gwyddion39 commented:
@egorpushkin A downside to splitting up the package and also to building a directory structure is that one of the dlls is shared between android and ios. It wouldn't be a huge deal to copy it into both since it's really small at 11kb.

I'm thinking a reasonable file structure would be:
/lib/EnRouteApi/EnRouteApi.dll (shared)
/lib/EnRouteApiAndroid/EnRouteApi.Android.dll
/lib/EnRouteApiiOS/EnRouteApi.Android.dll
/etc/version.properties

Possible pro for splitting up the packages is that the iOS library unzipped is quite large. So if for some reason you only wanted the Android one you would get a much smaller package. That said, since these are for Xamarin I think it's unlikely someone would want only one and not both.

egorpushkin commented:
@gwyddion39 Sorry for sending you down this route. We will be deprecating the use of Dropbox. S3 is sufficient. I totally blanked that our Jenkins is already configured to upload to S3. Lets change the path to:
`builds.glympse.com/enroute/sdk/xamarin/${JOB_NAME}/${BUILD_NUMBER}`

egorpushkin commented:
PAC

gwyddion39 commented:
@egorpushkin Done

Removed the dropbox upload script. Cleaned the dropbox config file from the build server. Changed the S3 target path in the build configuration.

egorpushkin commented on docs/changelog.md:
```diff
@@ -0,0 +1,6 @@
+# Glympse Enroute Xamarin Client SDK Changelog
```
We will maintain it in web-dgc repo.

gwyddion39 replied:
Can pull it from there in the build script. Keep in mind this makes the build script in a public repo refer to a private repo.

egorpushkin commented on version.properties:
```diff
@@ -0,0 +1 @@
+XAMARIN_SDK_VERSION=1.0.0
```
Should we have version in the source code (like we do in native SDKs) so it can be baked into the binary?

gwyddion39 replied:
Yes, this was the simplest solution for the moment since I needed to access the version number from Jenkins. Putting the version in source code, we'll still need to extract it and create this file.

egorpushkin commented:
@gwyddion39 
Remove changelog
LGTM
